src/components/OnboardingForm.tsx

```diff
@@ -126,12 +122,37 @@ export const OnboardingForm = () => {
     setFieldErrors(errors);
 
     if (Object.keys(errors).length === 0) {
-      setFormIsValid(true);
+      return true;
+    }
+
+    if (isFromSubmitted) {
+      if (errors.email) {
+        var email: HTMLElement | void = document
+          .getElementById("email")
+          ?.focus();
+        return false;
+      }
+      if (errors.orgName) {
+        var orgName: HTMLElement | void = document
+          .getElementById("org-name")
+          ?.focus();
+        return false;
+      }
     }
+    return false;
+  };
+
+  useEffect(() => {
+    validateForm();
   }, [fieldValues, formIsDirty]);
```

Comment:
I am confused, you call this function but it will immediately return so i will assume that you don't need to call this function at all

Reply:
If we look at it more closely, then it is required. But yes, the way it is written looks a bit wired, I should have written it in a more elegant way, without overkilling the solution
and have only `Just validate the form when the user submits it`
I was just trying to make it more user-friendly 😅. Thank you for mentioning it here.